clevertap-core/src/main/java/com/clevertap/android/sdk/network/api/CtApi.kt

```diff
@@ -147,28 +140,18 @@ internal class CtApi(
             }
         }
 
-        if (isViewedEvent) {
-            if (spikyProxyDomain.isNotNullAndBlank()) {
-                return spikyProxyDomain!!
-            }
-        } else {
-            if (proxyDomain.isNotNullAndBlank()) {
-                return proxyDomain!!
-            }
+        val toCheckProxy = if (isViewedEvent) { spikyProxyDomain } else { proxyDomain }
+        if (toCheckProxy.isNotNullAndBlank()) {
+            return toCheckProxy!!
         }
 
         if (customHandshakeDomain.isNotNullAndBlank()) {
             return customHandshakeDomain!!
         }
 
-        if (isViewedEvent) {
-            if (cachedSpikyDomain.isNotNullAndBlank()) {
-                return cachedSpikyDomain!!
-            }
-        } else {
-            if (cachedDomain.isNotNullAndBlank()) {
-                return cachedDomain!!
-            }
+        val toCheckCached = if (isViewedEvent) { cachedSpikyDomain } else { cachedDomain }
+        if (toCheckCached.isNotNullAndBlank()) {
+            return toCheckCached!!
```

Comment:
Just a suggestion: We can drop the redundant !! mark if we define ```isNotNullAndBlank``` with a contract:
```kotlin
@OptIn(ExperimentalContracts::class)
fun String?.isNotNullAndBlank() : Boolean {
    contract { returns(true) implies (this@isNotNullAndBlank != null) }
    return isNullOrBlank().not()
}
```
It is still an experimental feature, but I think it is safe for this use case. It just tells the compiler that the function includes a null check.

Reply:
done.